sormas-backend/src/main/java/de/symeda/sormas/backend/common/BaseAdoService.java

```diff
@@ -202,6 +216,20 @@ public Predicate createChangeDateFilter(CriteriaBuilder cb, From<?, ADO> from, D
 		return createChangeDateFilter(cb, from, DateHelper.toTimestampUpper(date));
 	}
 
+	public Predicate createChangeDateFilter(CriteriaBuilder cb, From<?, ADO> from, Date date, String lastSynchronizedUuidSameTimestamp) {
+		if (lastSynchronizedUuidSameTimestamp == null || EntityDto.NO_LAST_SYNCED_UUID.equals(lastSynchronizedUuidSameTimestamp)) {
+			return createChangeDateFilter(cb, from, date);
+		} else {
+			Timestamp timestamp = DateHelper.toTimestampUpper(date);
+			Predicate predicate = cb.or(
+				cb.greaterThan(from.get(AbstractDomainObject.CHANGE_DATE), timestamp),
+				cb.and(
+					cb.equal(from.get(AbstractDomainObject.CHANGE_DATE), timestamp),
```

Comment:
From my understanding this is not correct. Instead we need to take the "TimestampLower" (so basically the date) and say that the change date of the ADO is between lower and upper.
Here it's sufficient to check wether it's greaterOrEqual timestampLower

Reply:
Date comparisons in the `CriteriaBuilder` have precision milliseconds, so this fulfills the requirement.
Edge cases are modeled in `TaskFacadeEjbTest.testArchivedTaskNotGettingTransfered`.

Reply:
Refactored.